From: Jo Van Bulck <jo.vanbulck@cs.kuleuven.be>
Date: Thu, 19 Jan 2017 17:00:15 +0100
Subject: ecc: Store EdDSA session key in secure memory.
Origin: https://git.gnupg.org/cgi-bin/gitweb.cgi?p=libgcrypt.git;a=commit;h=f9494b3f258e01b6af8bd3941ce436bcc00afc56
Bug-Debian-Security: https://security-tracker.debian.org/tracker/CVE-2017-9526

* cipher/ecc-eddsa.c (_gcry_ecc_eddsa_sign): use mpi_snew to allocate
session key.
--

An attacker who learns the EdDSA session key from side-channel
observation during the signing process, can easily revover the long-
term secret key. Storing the session key in secure memory ensures that
constant time point operations are used in the MPI library.

Signed-off-by: Jo Van Bulck <jo.vanbulck@cs.kuleuven.be>
---
 cipher/ecc-eddsa.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/cipher/ecc-eddsa.c
+++ b/cipher/ecc-eddsa.c
@@ -584,7 +584,7 @@ _gcry_ecc_eddsa_sign (gcry_mpi_t input,
   a = mpi_snew (0);
   x = mpi_new (0);
   y = mpi_new (0);
-  r = mpi_new (0);
+  r = mpi_snew (0);
   ctx = _gcry_mpi_ec_p_internal_new (skey->E.model, skey->E.dialect, 0,
                                      skey->E.p, skey->E.a, skey->E.b);
   b = (ctx->nbits+7)/8;
